src/Generator/VaryGenerator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Common\Plugin\Generator;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Generate a cache key and specify what headers you want to vary on.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class VaryGenerator implements CacheKeyGenerator
```

Comment:
Thank you for the review. 
I cannot make it final because PHP spec need to extend this class to be able to run the tests.